d2/src/main/java/com/linkedin/d2/xds/XdsClientImpl.java

```diff
@@ -867,29 +873,26 @@ public void run()
       for (ResourceType type : _resourceSubscribers.keySet())
       {
         Set<String> resources = new HashSet<>(getResourceSubscriberMap(type).keySet());
-        if (resources.isEmpty() && getWildcardResourceSubscriber(type) == null)
+        if (resources.isEmpty())
         {
           continue;
         }
-        ResourceType rewrittenType;
+
+        ResourceType rewrittenType = type;
         if (_subscribeToUriGlobCollection && type == ResourceType.D2_URI_MAP)
         {
           resources = resources.stream()
               .map(GlobCollectionUtils::globCollectionUrlForClusterResource)
               .collect(Collectors.toCollection(HashSet::new));
           rewrittenType = ResourceType.D2_URI;
         }
-        else
-        {
-          rewrittenType = type;
-        }
```

Comment:
I actually prefer using the uninitialized variable declaration like `ResourceType rewrittenType;` because that way the compiler will complain if it's unset, so you are forced to look at both branches. For example, if a branch is added after, and it doesn't set `rewrittenType` the code will not compile. It's just a nice feature that makes it harder to mess up. If you don't mind keeping the original, I'd appreciate it

Reply:
sure, reverted.